src/main/java/com/squareup/javapoet/AnnotatedTypeName.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.javapoet;
+
+import java.io.IOException;
+import java.util.List;
+
+public final class AnnotatedTypeName extends TypeName {
```

Comment:
It seems strange to have a public type with no factory methods in contrast to what the rest of the library does. We don't have something like
```java
public ParameterizedTypeName parameterized(List<TypeName> parameters) {
  // ...
}
```
on `TypeName` (maybe we should!).

At the very least I think a static factory is in order here which takes a `TypeName` base an varargs `AnnotationSpec`.